.ci/muon/run.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+# Fetch and build
+#   muon (a C reimplementation of the meson build system),
+#   samurai (a C reimplementation of the ninja build tool),
+# and then use both to build tinc.
+
+set -euo pipefail
+
+git_samurai=https://github.com/michaelforney/samurai
+git_muon=https://git.sr.ht/~lattis/muon
+prefix=/opt/tinc_muon
+
+header() {
+  echo >&2 '################################################################################'
+  echo >&2 "# $*"
+  echo >&2 '################################################################################'
+}
+
+header 'Try to make sure Python is missing'
+python --version && exit 1
+python3 --version && exit 1
+
+header 'Fetch and build samurai'
+
+git clone --depth=1 $git_samurai ~/samurai
+pushd ~/samurai
+make -j"$(nproc)"
+make install
+popd
+
+header 'Fetch and build muon'
+
+git clone --depth=1 $git_muon ~/muon
+pushd ~/muon
+./bootstrap.sh build
+./build/muon setup build
+samu -C build
```

Comment:
Ah but if samu is built in to muon, you need to do this:
```
./build/muon -C build samu
```
No need to build and install samurai separately this way.

Reply:
I think the current approach is not only simpler, but also more realistic.

You need ninja or samurai to build muon, so there's the chicken and egg problem.

And I had several issues with using samurai through muon:

1. `samu.so` isn't installed automatically, so you have to copy it manually and use `LD_LIBRARY_PATH` or add `/usr/local/lib` to dynamic linker lookup paths
1. then it failed to recompile a build directory which both the original separate samu and ninja worked fine with both before and after (just errors out with an ok message, but non-zero exit code).

The first one could probably be solved by static linking, but we run into more issues since `libcurl-dev` doesn't support it (and it's necessary to fetch wraps).

The second is more problematic. I'll see if it's reproducible, and report to upstream if it is.

I was very impressed with muon as a whole, but this particular feature leaves a taste of WIP.

Reply:
Ok, let's keep it like this then. I also think muon is already pretty decent; we don't have a trivial meson setup and it handles almost everything fine.

Reply:
The embedded samu is considered rather experimental and hacky by the muon author, that being said, it should be getting built as a static library and statically linked into muon itself -- the samu subproject defines `default_options: ['default_library=static']`. So there should be no .so to install.

Reply:
Anyway, having a standalone samu is not unreasonable... :D